src/lib/y2storage/disk_analyzer.rb

```diff
@@ -254,105 +255,131 @@ def release_name(filesystem)
     end
 
     # Find disk devices that are suitable for installing Linux.
-    # Put any USB disks to the end of that array.
     #
     # @return [Array<Disk>] candidate disks
     def find_candidate_disks
-      @installation_disks = find_installation_disks
-
-      usb_disks, non_usb_disks = devicegraph.disk_devices.partition(&:usb?)
-
-      # Try with non-USB disks first.
-      disks = remove_installation_disks(non_usb_disks)
-      return disks unless disks.empty?
-
-      log.info("No non-USB candidate disks left after eliminating installation disks")
-      log.info("Trying with USB disks")
-      disks = remove_installation_disks(usb_disks)
-      return disks unless disks.empty?
-
-      # We don't want to install on our installation disk if there is any other way.
-      log.info("No candidate disks left after eliminating installation disks")
-      log.info("Trying with non-USB installation disks")
-      disks = @installation_disks.select { |d| !d.usb? }
-      return disks unless disks.empty?
-
-      log.info("Still no candidate disks left")
-      log.info("Trying with installation disks out of sheer desperation")
-      @installation_disks
-    end
-
-    # Find disk devices that look like the current installation medium
-    # (the medium we just booted from to start the installation).
-    #
-    # This should be limited because some architectures (s/390) tend
-    # to have a large number of disks, and checking if a disk is an
-    # installation disk involves mounting and unmounting each partition
-    # on that disk.
-    #
-    # @return [Array<Disk>]
-    def find_installation_disks
-      usb_disks, non_usb_disks = devicegraph.disk_devices.partition(&:usb?)
-      disks = usb_disks + non_usb_disks
-      disks = disks.first(DEFAULT_CHECK_LIMIT)
-      disks.select { |d| installation_disk?(d) }
-    end
-
-    # Check if a disk is our installation disk - the medium we just booted
-    # and started the installation from. This will check all filesystems on
-    # that disk.
-    #
-    # @param disk [Disk] device to check
-    # @return [Boolean] 'true' if the disk is an installation disk
-    def installation_disk?(disk)
-      log.info("Checking if #{disk.name} is an installation disk")
-      # Check if there is a filesystem directly on the disk (without partition table).
-      # This is very common for installation media such as USB sticks.
-      return installation_volume?(disk) if disk.partition_table.nil?
-
-      disk.partitions.each do |partition|
-        if NO_INSTALLATION_IDS.include?(partition.id)
-          log.info "Skipping #{partition} (ID #{partition.id.inspect})"
-          next
-        elsif installation_volume?(partition)
-          return true
-        end
-      end
+      devicegraph.disk_devices.select { |d| candidate_disk?(d) }
```

Comment:
So we are not longer discarding USB disks if a non-USB one is present? TBH, I never liked that idea much... but I don't feel we are in the right timeframe to change that behavior. It's not really related to the main topic (fixing detection of installation media).

Reply:
Do you think it is dangerous? I was thinking about it, and I would expect to have all devices (usb or not) available. But anyway, if this change is too big, I can go back to the old behavior.